iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -14,126 +15,104 @@ public class ClientPropertyCollection : PayloadCollection
-        /// <param name="propertyValue">The value of the property to add or update.</param>
-        /// <param name="statusCode"></param>
-        /// <param name="version"></param>
-        /// <param name="description"></param>
-        public void Add(string propertyName, object propertyValue, int statusCode, long version, string description = default)
-            => Add(null, propertyName, propertyValue, statusCode, version, description);
-
-        /// <inheritdoc path="/exception['ArgumentException']" cref="AddInternal(IDictionary{string, object}, string, bool)" />
-        /// <inheritdoc path="/seealso" cref="AddInternal(IDictionary{string, object}, string, bool)" />
-        /// <summary>
-        /// Adds a writable property to the collection.
-        /// </summary>
         /// <remarks>
-        /// This method will use the <see cref="PayloadSerializer.CreateWritablePropertyResponse(object, int, long, string)"/> method to create an instance of <see cref="IWritablePropertyResponse"/> that will be properly serialized.
+        /// If the collection already has a key matching a property name supplied this method will throw an <see cref="ArgumentException"/>.
+        /// <para>
+        /// When using this as part of the writable property flow to respond to a writable property update you should pass in the value
+        /// as an instance of <see cref="PayloadSerializer.CreateWritablePropertyResponse(object, int, long, string)"/>
+        /// to ensure the correct formatting is applied when the object is serialized.
+        /// </para>
+        /// <para>
+        /// This method directly adds the supplied <paramref name="properties"/> to the collection.
+        /// For component-level properties, either ensure that you include the component identifier markers {"__t": "c"} as a part of the supplied <paramref name="properties"/>,
+        /// or use the <see cref="AddComponentProperties(string, IDictionary{string, object})"/> convenience method instead.
+        /// For more information see <see href="https://docs.microsoft.com/en-us/azure/iot-pnp/concepts-convention#sample-multiple-components-read-only-property"/>.
+        /// </para>
         /// </remarks>
-        /// <exception cref="ArgumentNullException"><paramref name="propertyName"/> is <c>null</c> </exception>
-        /// <param name="propertyName">The name of the property to add or update.</param>
-        /// <param name="propertyValue">The value of the property to add or update.</param>
-        /// <param name="statusCode"></param>
-        /// <param name="version"></param>
-        /// <param name="description"></param>
-        /// <param name="componentName"></param>
-        public void Add(string componentName, string propertyName, object propertyValue, int statusCode, long version, string description = default)
+        /// <param name="properties">A collection of properties to add.</param>
+        public void Add(IDictionary<string, object> properties)
```

Comment:
I want to highlight this API that allows the user to add a dictionary directly to the collection.
For component-level properties, we recommend using the `AddComponentProperties(string, IDictionary{string, object})` convenience method instead, so that they don't need to ensure that the component-identifiers "__t": "c" are a part of the collection.